cmd/commands/app.go

```diff
@@ -227,3 +239,89 @@ func getCommitMsg(opts *AppCreateOptions) string {
 	}
 	return commitMsg
 }
+func NewAppListCommand() *cobra.Command {
+	var (
+		projectName   string
+	//	appListOpts   *application.
+		cloneOpts *git.CloneOptions
+	)
+
+	cmd := &cobra.Command{
+		Use:   "list [ENV_NAME]",
+		Short: "List all applications in an environment",
+		Example: util.Doc(`
+# To run this command you need to create a personal access token for your git provider,
+# and have a bootstrapped GitOps repository, and provide them using:
+	
+		export GIT_TOKEN=<token>
+		export GIT_REPO=<repo_url>
+
+# or with the flags:
+	
+		--token <token> --repo <repo_url>
+		
+# Get list of applications from a remote repository
+	
+	<BIN> app list <project_name>
+`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				log.G().Fatal("must enter environment name")
+			}
+			projectName = args[0]
+
+
+			return RunAppList(cmd.Context(), &AppListOptions{
+				ProjectName:      projectName,
+				FS:           fs.Create(memfs.New()),
+				CloneOptions: cloneOpts,
+			})
+		},
+	}
+	cloneOpts, err := git.AddFlags(cmd)
+	util.Die(err)
+
+	return cmd
+}
+
+func RunAppList(ctx context.Context, opts *AppListOptions) error {
+
+	var (
+		err error
+	)
+
+	log.G().WithFields(log.Fields{
+		"repoURL":  opts.CloneOptions.URL,
+		"revision": opts.CloneOptions.Revision,
+	}).Debug("starting with options: ")
+
+	// clone repo
+	log.G().Infof("cloning git repository: %s", opts.CloneOptions.URL)
+	_, opts.FS, err = opts.CloneOptions.Clone(ctx, opts.FS)
+	if err != nil {
+		return err
+	}
+
+	log.G().Infof("using revision: \"%s\", installation path: \"%s\"", opts.CloneOptions.Revision, opts.CloneOptions.RepoRoot)
+	if !opts.FS.ExistsOrDie(store.Default.BootsrtrapDir) {
+		log.G().Fatalf("Bootstrap folder not found, please execute `repo bootstrap --installation-path %s` command", opts.CloneOptions.RepoRoot)
+	}
+
+	envExists := opts.FS.ExistsOrDie(opts.FS.Join(store.Default.ProjectsDir, opts.ProjectName+".yaml"))
+	if !envExists {
+		log.G().Fatalf(util.Doc(fmt.Sprintf("env '%[1]s' not found, please execute `<BIN> env create %[1]s`", opts.ProjectName)))
+	}
+
+	log.G().Debug("repository is ok")
+
+	// get all apps beneath kustomize <env>\overlayes
+
+	matches, err := billyUtils.Glob(opts.FS, fmt.Sprintf("/kustomize/*/overlays/%s", opts.ProjectName))
+
+	for _, appName := range matches {
+
+		log.G().Info(strings.Split(appName, "/")[2])
```

Comment:
I would have liked to see something more than just an app name, but maybe that's enough for now

Reply:
for sure